src/JellyfishClient.ts

```diff
@@ -299,13 +315,25 @@ export class JellyfishClient<PeerMetadata, TrackMetadata> extends (EventEmitter
       this.emit("disconnected");
     });
     this.webrtc?.on("endpointAdded", (endpoint: Endpoint) => {
-      this.emit("peerJoined", endpoint);
+      if (endpoint.type === "webrtc") {
+        this.emit("peerJoined", endpoint);
+      } else {
+        this.emit("componentAdded", endpoint);
```

Comment:
I don't like this approach, it's like asking for bugs after jellyfish will support more and more protocols in the future.
I'm not sure though how to make it better. The different events for different protocols I can understand but that's also something that mby could be done better. In my opinion at the very least this "if ... else ..." should be wrapped in somewhat generic function that takes care of distinguishing between protocols and takes an event type as a parameter, then it can look up proper event name in some table. Wdyt?